docs/blueprints/coupon-codes.md

```diff
@@ -0,0 +1,70 @@
+# Coupon Codes 
+
+## Abstract
+
+Enable users to apply a promotional code to their account so that they can automatically receive the free credit associated with that code.
+
+Summary: When Sales/Marketing whats to run a promotion or hand out a specific amount of free credit to a group of people, we want to be able to generate a coupon code that a user could enter into their account, so that specific free credits and expiration of the credits would be automatically applied to that user’s account.
+
+## Design
+
+There are 2 high level aspects of this design:
+
+### Generating Coupon Codes
+
+This does not actually require any code changes - we can use Stripe's existing [promo codes functionality](https://stripe.com/docs/billing/subscriptions/discounts/codes). New promotional coupons can be created from the Stripe UI, then the codes can be delivered to customers through marketing.
+
+It is the responsibility of the individual creating a new coupon to ensure that its value is prefereable to the free tier coupon - we don't want the free tier coupon to be replaced with an inferior coupon.
+
+### Applying a Coupon Code to a User
+
+On account creation or from the payments panel, a user should be able to insert a coupon code and immediately have the corresponding coupon applied to their account. Because Stripe only allows for one coupon per customer, the default free tier coupon will be replaced with this new coupon.
+
+When a promotional coupon expires, we must apply the default free tier coupon to the customer.
+
+## Rationale
+
+The design below basically makes a call to the Stripe API when a user attempts to apply a coupon code. Because Stripe handles coupons on its own, there is no need for us to make any additional changes to the database or invoice generation logic.
+
+## Implementation
+
+### Service
+
+The service layer will communicate with the Stripe API to apply a coupon code to a user. Since there is very little functionality added, this can probably just be added to the stripecoinpayments service.
```

Comment:
I'm not a big fan of this entanglement, however, the issue is probably in the package names rather than location of things itself. I suspect a design how databases are handled could end up organizing things slightly better.

Note, you don't need to make changes due to this note.